### PR TITLE
[prometheusreceiver] improve metrics builder performance

### PR DESCRIPTION
**Pull request by newly12**
**Description:** <Describe what has changed.>

improve metrics builder performance

**Link to tracking Issue:** <Issue number if applicable>

**Testing:** <Describe what testing was performed and which tests were added.>

**Documentation:** <Describe the documentation added.>

### CODE REVIEW

File: receiver/prometheusreceiver/internal/otlp_metricsbuilder_test.go [hunk 1]
```diff
@@ -1036,6 +1038,7 @@ func Test_OTLPMetricBuilder_histogram(t *testing.T) {
 }
 
 func Test_OTLPMetricBuilder_summary(t *testing.T) {
+	// fix
```
Comment (dashpole) [inline receiver/prometheusreceiver/internal/otlp_metricsbuilder_test.go:1041]: What needs to be fixed here?

Comment (newly12) [reply receiver/prometheusreceiver/internal/otlp_metricsbuilder_test.go]: sorry it was mistakenly added. had just removed it.

**Review by dashpole — APPROVED** [review]

**Review by codeboten — APPROVED** [review]

**Comment by Mario-Hofstaetter** [comment]
> Improve performance of metrics builder

What exactly does this change improve? CPU load? Memory utilization? Latency?

**Comment by dashpole** [comment]
I had the same question initially, and had to go look at the MoveTo and CopyTo functions.  The answer is: probably a small amount of all of the above.  It avoids one copy for each metric, so it should save a few CPU cycles, and also doesn't have to allocate an additional metric for the copy.  I'll take a quick look at your issue.

**Pull request merged**